app/src/main/java/com/xinto/opencord/ui/component/channel/list/ChannelsList.kt

```diff
@@ -0,0 +1,63 @@
+package com.xinto.opencord.ui.component.channel.list
+
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.LocalAbsoluteTonalElevation
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.xinto.opencord.ui.viewmodel.ChannelsViewModel
+
+@Composable
+fun ChannelsList(
+    onChannelSelect: () -> Unit,
+    viewModel: ChannelsViewModel,
```

Comment:
if it's in the components folder, it shouldn't accept a viewmodel. components are meant to be reusable across the whole app and not only, viewmodels make that impossible.

Reply:
consider moving screen-related stuff to widgets folder instead

Reply:
or rather instead of widgets, move them to screens folder and make subfolders for all screens

Reply:
this applies to all further screen-related stuff that's in the components folder but I'm lazy to comment on all of them

Reply:
what about `panel` instead of widget
I have a personal vendetta against widget

Reply:
lmao sure why not, but IMO this structure would be good: `/ui/screen/{screen name}/{state}`. For example:
- `/ui/screen/home/Loaded.kt`
- `/ui/screen/home/Loading.kt`
- `/ui/screen/home/Error.kt`

or even:
- `/ui/screen/home/HomeScreenLoaded.kt`
- `/ui/screen/home/HomeScreenLoading.kt`
- `/ui/screen/home/HomeScreenError.kt`

Reply:
we can also further improve this by adding some extra stuff, for example:
- `/ui/screen/home/HomeScreenLoaded.kt`
- `/ui/screen/home/HomeScreenLoading.kt`
- `/ui/screen/home/HomeScreenError.kt`
- `/ui/screen/home/HomeScreenBottombar`
- `/ui/screen/home/HomeScreenTopbar`
- `/ui/screen/home/HomeScreenThatOneOverlyComplicatedWidget`

Reply:
Well
1. your example, Home doesn't have load states
2. if you're referring to channel list, it's not a screen and shouldn't be labeled as such

Reply:
well it's technically a screen since it has its own viewmodel and stuff

Reply:
the only reason it's not *actually* a screen is because of overlapping panels but I still consider them as screens because of viewmodel and states

Reply:
nopppp
no scaffold&app bar = not screen

Reply:
:skull:

Reply:
That's not how screens work.
Scaffolds are actually supposed to be the root of the app per material specs (the bottombar should always be visible, as well as the topbar)

Reply:
if a component has a viewmodel, then it's definitely a screen. viewmodels are there to provide sceren-specific functionality to components. all screens are components but not all components are screens.

Reply:
well material specs are deranged anyways :trollface: we're already not subscribing to all the specs
I still wouldn't call that a screen
there should only be one "screen" visible at once otherwise it doesn't make sense 

and no one actually uses one scaffold per app
it's stupid

Reply:
I still think that if a component has a viewmodel, then it's a screen. If not, then we can describe a screen by other definitions, for example, by the fact that screens are fundamentally different from each other. The channels list has nothing to do with Guilds list, chat panel or member panel, hence it's a different screen. If you still don't like that, then we can create a panels package for the home screen, where individual panels would have their components listed together, not interfering with other panels. An example would be:
- `/ui/screen/home/panel/channels/ChannelListStateLoaded`
- `/ui/screen/home/panel/channels/ChannelListStateLoading`
- `/ui/screen/home/panel/channels/ChannelListStateError`
- `/ui/screen/home/panel/channels/ChannelListTopBar`
- `/ui/screen/home/panel/channels/ChannelListSomethingElse`

Reply:
why is it weird? we need to scope panels to screens somehow

Reply:
panels are going to be used by specific screens only, there's no need to move them to high levels

Reply:
do we?
i dont really see a reason for doing so
i mean components arent scoped either to screens

Reply:
components aren't scoped because they are meant to be reusable across the whole app. if a component is not meant to be reusable across the whole app then it's best if we put it alongside the specific screen.

Reply:
fair